packages/x-data-grid/src/hooks/core/useGridProps.ts

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+import type { DataGridProps } from '../../models/props/DataGridProps';
+import type { GridPrivateApiCommon } from '../../models/api/gridApiCommon';
+import type { GridStateInitializer } from '../utils/useGridInitializeState';
+
+type Props = Pick<DataGridProps, 'rowCount'>;
+
+export const propsStateInitializer: GridStateInitializer<Props> = (state, props) => {
+  return {
+    ...state,
+    props: {
+      rowCount: props.rowCount,
+    },
+  };
+};
+
+export const useGridProps = <PrivateApi extends GridPrivateApiCommon>(
+  apiRef: React.MutableRefObject<PrivateApi>,
+  props: Props,
+) => {
+  React.useEffect(() => {
+    apiRef.current.setState((state) => ({
+      ...state,
+      props: {
+        rowCount: props.rowCount,
```

Comment:
I think we have to make `rowCount` part of the state for the server-side data source (a controlled state) as we need to update it internally based on different factors, I wanted to make it part of the pagination state, i.e. `pagination.rowCount` which should fix the page count selector as well.
[PR](https://github.com/mui/mui-x/pull/12317/files#diff-9a0e68c704ca6f0429e6db360d5133bc084fa1dbe9dc4323da03fe001a929beb) / [file](https://github.com/mui/mui-x/blob/b83276c3155e4a2268080598f977945a4130a3b7/packages/x-data-grid/src/hooks/features/pagination/useGridRowCount.ts) where I tried to do it.

Would it be useful to still keep this prop in state?

Reply:
Ah ok. Seems like it would make the selector work. Do you want to handle this issue in your PR or should we merge this as it is and you pick it up in your PR?

Reply:
Whatever you prefer, ideally it'd be better to do it in my PR (I'll most likely extract the independent changes in separate PRs which will allow to deliver them sooner)

We could keep this PR open though, if we have another prop to make part of state we can update the PR and merge it.

Reply:
I'll let you handle the issue then. I'll close, we can reopen anytime.